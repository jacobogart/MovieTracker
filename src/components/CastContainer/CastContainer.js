import React from 'react';
import CastCard from '../CastCard/CastCard';

export const CastContainer = (props) => {
  let castCards;

  if (props.cast) {
    castCards = props.cast.map(person => <CastCard data={person} />);
  } else {
    castCards = <p>Loading...</p>
  }

  return (
    <section className="more-details-bottom">
      <h2>Cast</h2>
      <div className="flex-wrapper">
        <div className="CastContainer">
          {castCards}
        </div>
      </div>
    </section>
  );
}
