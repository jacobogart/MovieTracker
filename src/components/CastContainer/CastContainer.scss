.more-details-bottom {
  padding: 0 225px;

  h2 {
    color: white;
    font-size: 1.8em;
    text-align: left;
    margin: 40px 0 20px 0;
  }

  .flex-wrapper {
    display: flex;
    justify-content: center;
  }

  .CastContainer {
    display: flex;
    flex-wrap: nowrap;
    height: 325px;
    margin: 0 0 80px 0;
    overflow-x: auto;
  }
}