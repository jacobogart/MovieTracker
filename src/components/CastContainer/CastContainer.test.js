import { shallow } from "enzyme";
import React from "react";
import { CastContainer } from "./CastContainer";

const mockCastMember = {
  name: "Chris Evans",
  character: "Captain America",
  headshot: "fsdfasdfa.png",
  id: 1
};

const mockCast = [mockCastMember, mockCastMember]

describe("CastCard", () => {
  it("should match the snapshot", () => {
    let wrapper = shallow(<CastContainer cast={mockCast} />);
    expect(wrapper).toMatchSnapshot();
  });
});
