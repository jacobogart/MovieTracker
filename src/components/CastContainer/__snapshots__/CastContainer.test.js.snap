// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CastCard should match the snapshot 1`] = `

<section
  className="more-details-bottom"
>
  <h2>
    Cast
  </h2>
  <div
    className="flex-wrapper"
  >
    <div
      className="CastContainer"
    >
      <CastCard
        data={
          Object {
            "character": "Captain America",
            "headshot": "fsdfasdfa.png",
            "id": 1,
            "name": "Chris Evans",
          }
        }
      />
      <CastCard
        data={
          Object {
            "character": "Captain America",
            "headshot": "fsdfasdfa.png",
            "id": 1,
            "name": "Chris Evans",
          }
        }
      />
    </div>
  </div>
</section>
`;
